apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx

```diff
@@ -265,6 +265,25 @@ export const StepBasedFlow: FunctionComponent<AuthenticationFlowPropsInterface>
         const steps: AuthenticationStepInterface[] = [ ...authenticationSteps ];
 
         const hasStepsWithOptions: boolean = steps.some((step: AuthenticationStepInterface) => !isEmpty(step.options));
+        let hasFoundTheSubjectIdentifiableStep: boolean = false;
+
+        authenticationSteps.forEach((step: AuthenticationStepInterface) => {
+            step.options.map((option: AuthenticatorInterface) => {
+                const isSubjectIdentifiableAuthenticator: boolean = ![
+                    IdentityProviderManagementConstants.TOTP_AUTHENTICATOR,
+                    IdentityProviderManagementConstants.EMAIL_OTP_AUTHENTICATOR,
+                    IdentityProviderManagementConstants.SMS_OTP_AUTHENTICATOR,
+                    IdentityProviderManagementConstants.IDENTIFIER_FIRST_AUTHENTICATOR
+                ].includes(option.authenticator);
+
+                if (isSubjectIdentifiableAuthenticator) {
+                    if (step.id !== subjectStepId && !hasFoundTheSubjectIdentifiableStep) {
+                        setSubjectStepId(step.id);
+                        hasFoundTheSubjectIdentifiableStep = true;
+                    }
+                }
+            });
+        });
```

Comment:
It would be better if we can break the foreach loop once the subject identifiable step is found, wouldn't it? In such case, we can use [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some), can't we?

```suggestion
            const subjectIdentifiableStep: AuthenticationStepInterface = authenticationSteps.some((step: AuthenticationStepInterface) => {
                     // returns the step with subject identifiable authenticator included
                     return step.options.some((option: AuthenticatorInterface) => {
                          const isSubjectIdentifiableAuthenticator: boolean = ![
                                 IdentityProviderManagementConstants.TOTP_AUTHENTICATOR,
                                 IdentityProviderManagementConstants.EMAIL_OTP_AUTHENTICATOR,
                                 IdentityProviderManagementConstants.SMS_OTP_AUTHENTICATOR,
                                 IdentityProviderManagementConstants.IDENTIFIER_FIRST_AUTHENTICATOR
                          ].includes(option.authenticator);

                         return isSubjectIdentifiableAuthenticator;
                    });
             });

            If (subjectIdentifiableStep.id !== subjectStepId) {
               setSubjectStepId(subjectIdentifiableStep.id)
            }
```

PS: untested code, posted just as a suggestion :)